homeassistant/components/homematicip_cloud/hap.py

```diff
@@ -133,14 +147,20 @@ def async_update(self, *args, **kwargs):
         are set to unavailable.
         """
         if not self.home.connected:
+            # Hass is still connected to the HMIP cloud. The HAP is not.
+
             _LOGGER.error(
-                "HMIP access point has lost connection with the cloud")
+                "HMIP access point has lost connection with the cloud"
+            )
             self._accesspoint_connected = False
             self.set_all_to_unavailable()
-        elif not self._accesspoint_connected:
+
+        elif self.home.connected and not self._accesspoint_connected:
```

Comment:
Why change this line? First logic check seems already covered by previous if check.